R/stan_module.R

```diff
@@ -0,0 +1,93 @@
+#' StanAll class object
+#' @slot functions Character, the functions part of a stan model
+#' @slot data Character, the data part of a stan model
+#' @slot parameters Character, the parameters part of a stan model
+#' @slot transformed_parameters Character, the transformed_parameters part of a stan model
+#' @slot model Character, the model part of a stan model
+#' @slot prior List, the priors part of a stan model
+#' @slot generated_quantities Character, the generated_quantities part of a stan model
+#' @slot includes Character
+#' @slot inits List with the initial values of the stan model.
+#' @exportClass StanModule
+StanModule <- setClass(
+    "StanModule",
+    representation(
+        functions = "character",
+        data = "character",
+        parameters = "character",
+        transformed_parameters = "character",
+        prior = "list",
+        generated_quantities = "character",
+        inits = "list"
+    )
+)
+
+
+
+#' source_stan_part exports a .stan file as a character
+#' @param filename The name of the .stan file in the directory of the package.
+#' @export
+source_stan_part <- function(filename) {
+    if (file.exists(filename)) {
+        absolute_filename <- filename
+    } else {
+        absolute_filename <- system.file(
+            "stanparts",
+            filename,
+            package = "jmpost"
+        )
+    }
+
+    readLines(absolute_filename)
```

Comment:
Sorry this logic doesn't make sense to me.   If the user were to provide the stan code as a string i.e.

```
parameters = "
    real beta;
    real gamma;
    real sigma;
"
```

This would error because line 34 would try and find it as as a system file ?

Reply:
Indeed, allowing the user to add the code as a text could be helpful. Although I will keep the option of the `system.file` as well. It is important because we want to locate the examples we are going to provide along with the package.